cli/flox-rust-sdk/src/models/environment/mod.rs

```diff
@@ -254,14 +254,6 @@ pub enum ConcreteEnvironment {
 }
 
 impl ConcreteEnvironment {
-    pub fn into_dyn_environment(self) -> Box<dyn Environment> {
-        match self {
-            ConcreteEnvironment::Path(path_env) => Box::new(path_env),
-            ConcreteEnvironment::Managed(managed_env) => Box::new(managed_env),
-            ConcreteEnvironment::Remote(remote_env) => Box::new(remote_env),
-        }
-    }
```

Comment:
The issue didnt name it specifically but we should also remove `dyn_environment_ref[_mut]` since `& [mut] ConcreteEnvironment` is already an `&[mut] dyn Environment` now.